vulnfeeds/git/versions_test.go

```diff
@@ -0,0 +1,53 @@
+package git
+
+import (
+	"testing"
+)
+
+func TestVersionToCommit(t *testing.T) {
+	var cache RepoTagsCache
+	cache = make(RepoTagsCache)
+
+	tests := []struct {
+		description    string
+		inputRepoURL   string
+		cache          RepoTagsCache
+		inputVersion   string
+		expectedResult string
+		expectedOk     bool
+	}{
+		{
+			description:    "An exact match",
+			inputRepoURL:   "https://github.com/ARMmbed/mbedtls",
+			cache:          cache,
+			inputVersion:   "3.0.0",
+			expectedResult: "ecf535f6cb70c97beb7c44dceadc14423086ca0d",
+			expectedOk:     true,
+		},
+		{
+			description:    "A fuzzy version match",
+			inputRepoURL:   "https://gitlab.com/gitlab-org/gitlab",
+			cache:          cache,
+			inputVersion:   "12.0",
+			expectedResult: "1d73db751369085d05d939c798e7da086646696c",
+			expectedOk:     true,
```

Comment:
Can we add a false case as well?

Reply:
Done in [4d32497](https://github.com/google/osv.dev/pull/1151/commits/4d3249741636a1311a19d70f2af571f9f69e2e70)